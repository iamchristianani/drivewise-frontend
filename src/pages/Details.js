import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams, NavLink, useNavigate } from 'react-router-dom';
import './style/Details.css';
import { BiLeftArrow } from 'react-icons/bi';
import { IoIosArrowDropright } from 'react-icons/io';
import { getDetailsAction } from '../redux/cars/carDetails';
import Navbar from '../components/Navbar';

const Details = () => {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.details);
  const { carId } = useParams();
  const navigate = useNavigate();

  useEffect(() => {
    dispatch(getDetailsAction(carId));
  }, [dispatch, carId]);

  const reserveBtnClick = () => {
    navigate('/reservation_form');
  };

  return (
    <div className="page_container">
      <Navbar />
      <div className="details_container">
        <div className="inner-details">
          <div className="all-details">
            <div className="image-box">
              <img src={data.image} alt={data.model} />
            </div>
            <div className="all-text-box">
              <div className="title-x-desc">
                <h2>{data.make}</h2>
                <p className="details-model">
                  (
                  {data.model}
                  )
                </p>
                <p className="desc-text">{data.description}</p>
              </div>
              <div className="price-box">
                <p>Price:</p>
                <p className="details-amount">
                  $
                  {data.price}
                </p>
              </div>
              <div className="details-table-container">
                <table className="details-table">
                  <tbody>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Year</span>
                        <span className="table-right-text">{data.year}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Color</span>
                        <span className="table-right-text">{data.color}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Body Type</span>
                        <span className="table-right-text">{data.body_type}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Engine Capacity</span>
                        <span className="table-right-text">{data.engine_capacity}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Engine Cylinders</span>
                        <span className="table-right-text">{data.engine_cylinders}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Turbo</span>
                        <span className="table-right-text">{typeof data.turbo === 'boolean' ? data.turbo.toString() : 'Not available'}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Fuel Type</span>
                        <span className="table-right-text">{data.fuel_type}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Doors</span>
                        <span className="table-right-text">{data.doors}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Seats</span>
                        <span className="table-right-text">{data.seats}</span>
                      </td>
                    </tr>
                    <tr className="details-table-row">
                      <td>
                        <span className="table-left-text">Mileage</span>
                        <span className="table-right-text">{data.mileage}</span>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <button type="button" className="details-reserve-btn" onClick={reserveBtnClick}>
                Reserve
                <IoIosArrowDropright className="circle-arrow" />
              </button>
            </div>
          </div>
        </div>
        <div className="back-button">
          <NavLink to="/">
            <button type="button" className="scroll left-scroll" aria-label="Scroll Left"><BiLeftArrow /></button>
          </NavLink>
        </div>
      </div>
    </div>
  );
};

export default Details;
