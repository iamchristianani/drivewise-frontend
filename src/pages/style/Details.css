.details_container {
  width: 100%;
}

.inner-details {
  display: flex;
  flex-direction: column;
  padding: 50px;
  width: 100%;
}

.all-details {
  display: flex;
  gap: 20px;
}

.image-box {
  width: 70%;
  height: auto;
}

.image-box img {
  width: 100%;
}

.all-text-box {
  width: 30%;
  display: flex;
  flex-direction: column;
}

.title-x-desc {
  text-align: right;
}

.desc-text {
  font-size: 12px;
}

.details-table-container {
  margin-top: 30px;
}

.details-model {
  font-size: 12px;
  margin: -5px 0 10px 0;
}

.details-table {
  width: 100%;
  font-size: 12px;
  border-collapse: collapse;
}

.table-left-text {
  padding: 0 0 0 10px;
}

.table-right-text {
  float: right;
  padding: 0 10px 0 0;
}

.details-table-row {
  width: 100%;
  height: 30px;
}

.details-table-row:nth-child(even) {
  background-color: #fff;
}

.details-table-row:nth-child(odd) {
  background-color: #f3f3f3;
}

.price-box {
  margin: 20px 0 0 0;
  text-align: right;
}

.details-amount {
  font-size: 40px;
}

.back-button {
  position: relative;
  margin-bottom: 50px;
}

.details-reserve-btn {
  width: fit-content;
  margin: 50px 0 0 0;
  display: flex;
  align-items: center;
  padding: 15px 15px;
  border: 0;
  border-radius: 30px;
  background: #97bf0f;
  color: #fff;
  align-self: flex-end;
  cursor: pointer;
}

.circle-arrow {
  font-size: 20px;
  margin: 0 0 0 20px;
}

@media (max-width: 600px) {
  .details_container {
    margin-top: 50px;
  }

  .inner-details {
    padding: 20px;
  }

  .all-details {
    flex-direction: column;
  }

  .image-box {
    width: 100%;
  }

  .all-text-box {
    width: 100%;
  }

  .back-button {
    margin: 30px 0 70px 0;
  }
}
