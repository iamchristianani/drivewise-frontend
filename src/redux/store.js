import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import carsReducer from './cars/cars';
import detailsReducer from './cars/carDetails';
import reservationsReducer from './reservations/reservations';
import authSlice from './loginByUsername';

const store = configureStore(
  {
    reducer: {
      authentications: authSlice,
      cars: carsReducer,
      details: detailsReducer,
      reservations: reservationsReducer,
    },
  },
  applyMiddleware(thunk),
);

export default store;
