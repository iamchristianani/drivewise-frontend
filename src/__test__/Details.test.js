import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Details from '../pages/Details';

describe('Home Test', () => {
  test('to render each car details: ', () => {
    const home = render(
      <Provider store={store}>
        <BrowserRouter>
          <Details />
        </BrowserRouter>
      </Provider>,
    );
    expect(home).toMatchSnapshot();
  });
});
