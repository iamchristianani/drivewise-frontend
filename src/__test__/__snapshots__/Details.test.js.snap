// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Test to render each car details:  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="page_container"
      >
        <nav>
          <a
            href="/"
          >
            <img
              alt="Drivewise"
              src="drivewise_logo.png"
            />
          </a>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars fa-lg "
            data-icon="bars"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-x not-active"
            data-icon="x"
            data-prefix="fas"
            data-testid="mobile-button"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z"
              fill="currentColor"
            />
          </svg>
          <ul
            class="mobile-menu "
            data-testid="mobile-menu"
          >
            <li
              class=""
            >
              <a
                href="/"
              >
                Cars
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/reservation_form"
              >
                Add Reservation
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/my_reservations"
              >
                My Reservations
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/new_car"
              >
                Add Car
              </a>
            </li>
            <li
              class=""
            >
              <a
                href="/delete_car"
              >
                Delete Car
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="details_container"
        >
          <div
            class="inner-details"
          >
            <div
              class="all-details"
            >
              <div
                class="image-box"
              >
                <img />
              </div>
              <div
                class="all-text-box"
              >
                <div
                  class="title-x-desc"
                >
                  <h2 />
                  <p
                    class="details-model"
                  >
                    (
                    )
                  </p>
                  <p
                    class="desc-text"
                  />
                </div>
                <div
                  class="price-box"
                >
                  <p>
                    Price:
                  </p>
                  <p
                    class="details-amount"
                  >
                    $
                  </p>
                </div>
                <div
                  class="details-table-container"
                >
                  <table
                    class="details-table"
                  >
                    <tbody>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Year
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Color
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Body Type
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Engine Capacity
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Engine Cylinders
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Turbo
                          </span>
                          <span
                            class="table-right-text"
                          >
                            Not available
                          </span>
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Fuel Type
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Doors
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Seats
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                      <tr
                        class="details-table-row"
                      >
                        <td>
                          <span
                            class="table-left-text"
                          >
                            Mileage
                          </span>
                          <span
                            class="table-right-text"
                          />
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <button
                  class="details-reserve-btn"
                  type="button"
                >
                  Reserve
                  <svg
                    class="circle-arrow"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M210.7 147.6c7.5-7.5 19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81.1-81.9-81.1c-7.6-7.4-7.6-19.6 0-27.2z"
                    />
                    <path
                      d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm32 0c0-47 18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="back-button"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <button
                aria-label="Scroll Left"
                class="scroll left-scroll"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18.464 2.114a.998.998 0 0 0-1.033.063l-13 9a1.003 1.003 0 0 0 0 1.645l13 9A1 1 0 0 0 19 21V3a1 1 0 0 0-.536-.886zM17 19.091 6.757 12 17 4.909v14.182z"
                  />
                </svg>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="page_container"
    >
      <nav>
        <a
          href="/"
        >
          <img
            alt="Drivewise"
            src="drivewise_logo.png"
          />
        </a>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bars fa-lg "
          data-icon="bars"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-x not-active"
          data-icon="x"
          data-prefix="fas"
          data-testid="mobile-button"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z"
            fill="currentColor"
          />
        </svg>
        <ul
          class="mobile-menu "
          data-testid="mobile-menu"
        >
          <li
            class=""
          >
            <a
              href="/"
            >
              Cars
            </a>
          </li>
          <li
            class=""
          >
            <a
              href="/reservation_form"
            >
              Add Reservation
            </a>
          </li>
          <li
            class=""
          >
            <a
              href="/my_reservations"
            >
              My Reservations
            </a>
          </li>
          <li
            class=""
          >
            <a
              href="/new_car"
            >
              Add Car
            </a>
          </li>
          <li
            class=""
          >
            <a
              href="/delete_car"
            >
              Delete Car
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="details_container"
      >
        <div
          class="inner-details"
        >
          <div
            class="all-details"
          >
            <div
              class="image-box"
            >
              <img />
            </div>
            <div
              class="all-text-box"
            >
              <div
                class="title-x-desc"
              >
                <h2 />
                <p
                  class="details-model"
                >
                  (
                  )
                </p>
                <p
                  class="desc-text"
                />
              </div>
              <div
                class="price-box"
              >
                <p>
                  Price:
                </p>
                <p
                  class="details-amount"
                >
                  $
                </p>
              </div>
              <div
                class="details-table-container"
              >
                <table
                  class="details-table"
                >
                  <tbody>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Year
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Color
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Body Type
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Engine Capacity
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Engine Cylinders
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Turbo
                        </span>
                        <span
                          class="table-right-text"
                        >
                          Not available
                        </span>
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Fuel Type
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Doors
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Seats
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                    <tr
                      class="details-table-row"
                    >
                      <td>
                        <span
                          class="table-left-text"
                        >
                          Mileage
                        </span>
                        <span
                          class="table-right-text"
                        />
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <button
                class="details-reserve-btn"
                type="button"
              >
                Reserve
                <svg
                  class="circle-arrow"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M210.7 147.6c7.5-7.5 19.8-7.5 27.3 0l95.4 95.7c7.3 7.3 7.5 19.1.6 26.6l-94 94.3c-3.8 3.8-8.7 5.7-13.7 5.7-4.9 0-9.9-1.9-13.6-5.6-7.5-7.5-7.6-19.7 0-27.3l79.9-81.1-81.9-81.1c-7.6-7.4-7.6-19.6 0-27.2z"
                  />
                  <path
                    d="M48 256c0 114.9 93.1 208 208 208s208-93.1 208-208S370.9 48 256 48 48 141.1 48 256zm32 0c0-47 18.3-91.2 51.6-124.4C164.8 98.3 209 80 256 80s91.2 18.3 124.4 51.6C413.7 164.8 432 209 432 256s-18.3 91.2-51.6 124.4C347.2 413.7 303 432 256 432s-91.2-18.3-124.4-51.6C98.3 347.2 80 303 80 256z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="back-button"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <button
              aria-label="Scroll Left"
              class="scroll left-scroll"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18.464 2.114a.998.998 0 0 0-1.033.063l-13 9a1.003 1.003 0 0 0 0 1.645l13 9A1 1 0 0 0 19 21V3a1 1 0 0 0-.536-.886zM17 19.091 6.757 12 17 4.909v14.182z"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
